### PR TITLE
Add alarm clusters support

### PR DESCRIPTION
**Pull request by elemoine**
This introduces support for alarm clusters. WIP at the moment.

### CODE REVIEW

File: heka/files/lua/common/gse.lua [hunk 1]
```diff
@@ -36,10 +36,10 @@ local ordered_clusters = {}
```
Comment (SwannCroiset) [inline heka/files/lua/common/gse.lua:36]: I would really like to have Lua unittests. What about the feature branch **mkxx** on legacy repository ?

Comment (elemoine) [reply heka/files/lua/common/gse.lua:36]: I understand your concerns. Thinking about it I am not sure creating an mkxx branch in fuel-plugin-lma-collector makes sense, but let's discuss it.

Comment (elemoine) [reply heka/files/lua/common/gse.lua:36]: https://github.com/tcpcloud/salt-formula-heka/pull/32 adds a script for running Lua tests, and all the Lua tests from fuel-plugin-lma-collector.

File: heka/_service.sls [hunk 2]
```diff
@@ -279,6 +283,60 @@ heka_{{ service_name }}_grain:
...
+{%- set policy = service_metadata.get('policy') %}
+{%- if policy %}
+/usr/share/lma_collector/common/gse_policies.lua:
+  file.managed:
+  - source: salt://heka/files/gse_policies.lua
+  - template: jinja
+  - mode: 640
+  - group: heka
+  - require:
+    - file: heka_{{ service_name }}_conf_dir
```
Comment (elemoine) [inline heka/_service.sls]: Change this, the file is not in the conf dir.

File: heka/_service.sls [hunk 3]
```diff
@@ -279,6 +283,60 @@ heka_{{ service_name }}_grain:
...
+      alarm_cluster: {{ alarm_cluster|yaml }}
+
+/usr/share/lma_collector/common/gse_{{ alarm_cluster_name|replace('-', '_') }}_topology.lua:
+  file.managed:
+  - source: salt://heka/files/gse_topology.lua
+  - template: jinja
+  - mode: 640
+  - group: heka
+  - require:
+    - file: heka_{{ service_name }}_conf_dir
```
Comment (elemoine) [inline heka/_service.sls]: Change this, the file is not in the conf dir.

File: heka/files/toml/filter/gse_alarm_cluster.toml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+type = "SandboxFilter"
+filename = "/usr/share/lma_collector/filters/gse_cluster_filter.lua"
+preserve_data = {{ alarm_cluster.preserve_data|default(False)|lower }}
+message_matcher = "(Type == 'heka.sandbox.afd_metric' || Type == 'heka.sandbox.gse_metric') && ({{ salt['heka_alarming.alarm_cluster_message_matcher'](alarm_cluster) }})"
+module_directory = "/usr/share/lma_collector/common;/usr/share/heka/lua_modules"
+ticker_interval = 1
+
+[gse_{{ alarm_cluster_name }}_filter.config]
+topology_file = "gse_{{ alarm_cluster_name|replace('-', '_') }}_topology"
+dimensions = '{{ salt['heka_alarming.dimensions'](alarm_cluster)|json }}'
```
Comment (elemoine) [inline heka/files/toml/filter/gse_alarm_cluster.toml]: Add `interval`, `warm_up_period`, etc. (all the config options supported by the `gse_cluster_filter`).

File: heka/meta/heka.yml [hunk 5]
```diff
@@ -108,6 +108,41 @@ remote_collector:
...
+            threshold: 0
+    - status: okay
+      trigger:
+        logical_operator: or
+        rules:
+          - function: count
+            arguments: [ okay ]
+            relational_operator: '>'
+            threshold: 0
+    - status: unknown
```
Comment (elemoine) [inline heka/meta/heka.yml]: Add the other built-in policies.

**Comment by elemoine** [comment]
This is now ready for review. What I have not tested up to now is clusters of clusters.

File: heka/files/lua/common/gse.lua (hunk 1 above)
Comment (elemoine) [reply heka/files/lua/common/gse.lua:36]: #32 is merged, and this PR changes some of the tests.

**Review by simonpasquier — APPROVED** [review]

**Pull request merged**